README.md

```diff
@@ -73,14 +73,55 @@ options (
 ### Parallel queries
 `parquet_fdw` also supports [parallel query execution](https://www.postgresql.org/docs/current/parallel-query.html) (not to confuse with multi-threaded decoding feature of `arrow`). It is disabled by default; to enable it run `ANALYZE` command on the table. The reason behind this is that without statistics postgres may end up choosing a terrible parallel plan for certain queries which would be much worse than a serial one (e.g. grouping by a column with large number of distinct values).
 
-### Experimental
+### Import
 
 `parquet_fdw` also supports [`IMPORT FOREIGN SCHEMA`](https://www.postgresql.org/docs/current/sql-importforeignschema.html) command to discover parquet files in the specified directory on filesystem and create foreign tables according to those files. It can be used as follows:
 
-```
-IMPORT FOREIGN SCHEMA "/path/to/directory"
-FROM SERVER parquet_srv
-INTO public;
+```sql
+import foreign schema "/path/to/directory"
+from server parquet_srv
+into public;
 ```
 
 It is important that `remote_schema` here is a path to a local filesystem directory and is double quoted.
+
+Another way to import parquet files into foreign tables is to use `import_parquet` or `import_parquet_explicit`:
+
+```sql
+create function import_parquet(
+    tablename   text,
+    schemaname  text,
+    servername  text,
+    userfunc    regproc,
+    args        jsonb,
+    options     jsonb)
+
+create function import_parquet_explicit(
+    tablename   text,
+    schemaname  text,
+    servername  text,
+    attrs       text[],
+    userfunc    regproc,
+    args        jsonb,
+    options     jsonb)
+```
+
+The only difference between `import_parquet` and `import_parquet_explicit` is that the latter allows to specify a set of attributes (columns) to import. `attrs` here is a 2-dimensional text array of the form `[['column1', 'type1'], ['column2', 'type2']]` (see the example below).
```

Comment:
I've never been a fan of using multi-dimensional arrays instead of hashes. IIRC Postgres source code favors passing names and types separately. That would require no additional explanations and has an advantage that we could use proper types for those arrays (`name` for the attribute names and `regtype` for the types), moving correctness check from out code into the postgres itself. So I'd change this function to pass two separate parameters, 
attnames []name, atttypes []regtype).

Reply:
It would be also nice to document why we are not incorporating this into the "create foreign table" interface (my understanding, based on prior slack discussions, is that we need to parse `ftoptions` every time we access a foreign table in order to figure out which parquet files to read, and adding to that a call to a function that may reach out to the filesystem may impact the performance.

Reply:
Hi @alexeyklyukin,

Thanks for the review.
1. More tests is always good : )

2. I'll try to build on mac os and will add necessary headers.

3. Yes, I can change the way of passing attname and types. I had doubts over which way is better so i ended up with 2d array. I don't have preference here.

4. When creating a foreign table we hypothetically could pass a function name and its argument as table option but then it would need to be executed everytime we query this table. That might be suboptimal as this function might need to scan filesystem first (which in turn could be a ceph cluster meaning probably higher latency). The idea was to be able to run function once at creation time finding all the paths and add them statically to the table options. (Actually I have another branch on github [[userfunc](https://github.com/adjust/parquet_fdw/tree/userfunc)] that implements running function on every query. I'll create a new PR after this one gets committed).

PS: I'm a bit occupied right now, will fix as soon as finish current work.

Reply:
Sure, running function on every query due to the limitations of postgres foreign table catalogs (namely, we cannot run the function ones and rewrite `ftopions` in an "approved" way) is suboptimal, my point is to document it. I'll add some docs and tests (and a header from Mac, thanks). I'll see if I can easily change the code to incorporate two arrays as well.

Reply:
I changed arguments list in `import_parquet_explicit()` in a way to pass two separate arrays (`attnames` and `attypes`) instead of one 2d array.